api-gateway/README.md

```diff
@@ -0,0 +1,51 @@
+# SaveOurTool gateway
+
+## Overview
+Here is a brief scheme of routing in cluster:
+1. The Ingress proxies requests to `api-gateway` (configured in `ingress.yml`);
+2. `api-gateway` forwards requests to Spring services as well as `save-frontend` (configured in `application.yml` and `WebSecurityConfig.kt`);
+3. `save-frontend` is a pod that has `nginx` running in it (configured in `nginx.conf`);
+4. All further routing is configured in `basicRouting` functional component, defined in `save-frontend` module.
+
+### 1. Ingress:
+The Ingress proxies requests to the API Gateway (configured in `ingress.yml`).
+
+In Kubernetes, an Ingress is a resource that manages external access to services within the cluster.
+It acts as a layer of traffic routing and load balancing.
+Existed ingress configuration defines two routing rules for the Ingress resource.
+ - paths that start with `/grafana/**` are forwarded to `grafana` service.
+ - all the other paths `/**` are forwarded to `api-gateway` spring service.
+
+### 2. `api-gateway`
+`api-gateway` uses Spring Cloud Gateway (configured in `application.yml` and `WebSecurityConfig.kt`).
+
+`api-gateway` acts as a request router and filter for `SaveOurTool`.
+It forwards requests to specific backend services based on their paths
+and performs some necessary filtering and header manipulation to ensure proper communication between the gateway and downstream services.
+The routes are defined based on specific URI paths:
+ - `/sec/**` is not forwarded anywhere but `api-gateway`'s `SecurityInfoController`,
+    which is responsible for `/sec/user` and `/sec/oauth-providers` endpoints
+ - `/api/sandbox` is forwarded to `save-sandbox`
+ - `/api/demo` is forwarded to `save-demo`
+ - `/api/cpg` is forwarded to `save-cpg-demo`
+ - `/api/**` is forwarded to `save-backend`
+ - `/neo4j/browser/**` is forwarded to `neo4j-browser`
```

Comment:
maybe move it to ingress layer? @akuleshov7 
why api-gateway routes it. it's a separate service as grafana

Reply:
Good point, will do it

Reply:
neo4j uses own authentications (like grafana)

Reply:
actually, neo4j browser is just browser and it does connect to neo4j database using credentials

gateway pass trough all requests to neo4j browser without authorization

Reply:
raised https://github.com/saveourtool/save-cloud/issues/2406

Reply:
Will do it, thanks